p2p/memp2p/memp2p_network.go

```diff
@@ -0,0 +1,159 @@
+package memp2p
+
+import (
+	"fmt"
+	"sync"
+
+	"github.com/ElrondNetwork/elrond-go-sandbox/p2p"
+)
+
+type MemP2PNetwork struct {
+	mutex       sync.RWMutex
+	peerIDs     []p2p.PeerID
+	peers       map[p2p.PeerID]*MemP2PMessenger
+	LogMessages bool
+	Messages    []p2p.MessageP2P
+}
+
+func NewMemP2PNetwork() *MemP2PNetwork {
+	var peerIDs []p2p.PeerID
+	var messages []p2p.MessageP2P
+	return &MemP2PNetwork{
+		mutex:       sync.RWMutex{},
+		peerIDs:     peerIDs,
+		peers:       make(map[p2p.PeerID]*MemP2PMessenger),
+		LogMessages: false,
+		Messages:    messages,
+	}
+}
+
+// ListAddresses provides the addresses of the known peers.
+func (network *MemP2PNetwork) ListAddresses() []string {
+	network.mutex.Lock()
+	addresses := make([]string, len(network.peerIDs))
+	i := 0
+	for _, peerID := range network.peerIDs {
+		addresses[i] = fmt.Sprintf("/memp2p/%s", peerID)
+		i++
+	}
+	network.mutex.Unlock()
+	return addresses
+}
+
+// ListAddresses provides the addresses of the known peers, except a specified one.
+func (network *MemP2PNetwork) ListAddressesExceptOne(peerIDToExclude p2p.PeerID) []string {
+	network.mutex.Lock()
+	resultingLength := len(network.peerIDs) - 1
+	if resultingLength <= 0 {
+		network.mutex.Unlock()
+		return []string{}
+	}
+	addresses := make([]string, resultingLength)
+	k := 0
+	for _, peerID := range network.peerIDs {
+		if peerID == peerIDToExclude {
+			continue
+		}
+		addresses[k] = fmt.Sprintf("/memp2p/%s", peerID)
+		k++
+	}
+	network.mutex.Unlock()
+	return addresses
+}
+
+// Peers provides a copy of its internal map of peers
+func (network *MemP2PNetwork) Peers() map[p2p.PeerID]*MemP2PMessenger {
+	network.mutex.RLock()
+	peersCopy := make(map[p2p.PeerID]*MemP2PMessenger)
+	for peerID, peer := range network.peers {
+		peersCopy[peerID] = peer
+	}
+	network.mutex.RUnlock()
+	return peersCopy
+}
+
+// PeersExcludingMessenger provides a copy of its internal map of peers, excluding a specific peer.
```

Comment:
Function name from comment is different from the actual function name

Reply:
Fixed.